packages/website/docs/state.md

```diff
@@ -67,51 +47,75 @@ const autocomplete = createAutocomplete({
 });
 ```
 
+You can also manually update the state using setters. It's useful to implement custom features on top of autocomplete.
+
+In the following example, when detecting a potential filter in the query, you're manually adding it to the list of active filters using [Context](context) and removing it from the query.
+
+```js
+const categories = ['Tops', 'Bottoms', 'Hats', 'Bags', 'Footwear'];
+
+const autocomplete = createAutocomplete({
+  // ...
+  onStateChange({ state, setQuery, setContext }) {
+    const { query, context } = state;
+    const match = categories.find((category) => query.endsWith(category));
+
+    if (match) {
+      const filters = [...context.filters, match];
+      const position = query.lastIndexOf(match);
+
+      setContext({ filters });
+      setQuery(query.substr(0, position));
+    }
+  },
+});
+```
+
 ## State
 
 ### `activeItemId`
 
 > `number | null` | defaults to `null`
 
-The highlighted item index.
+The highlighted item's index.
 
 ### `query`
 
 > `string` | defaults to `""`
 
-The query of the input.
+The value of the search input.
 
 ### `completion`
 
 > `string | null` | defaults to `null`
 
-The completion of the input.
+The completion of the query.
 
 ### `isOpen`
 
 > `boolean` | defaults to `false`
 
-Whether the panel is opened.
+Whether the panel is open or not.
 
 ### `collections`
 
 > `Collection[]` | defaults to `[]`
```

Comment:
What's your recommendation for documenting custom types? Should we link to their [source code](https://github.com/algolia/autocomplete.js/blob/d86a33a50adefa3b6996e80e1d3b407f0dab59be/packages/autocomplete-js/src/types/AutocompleteCollection.ts#L5-L8)?

Reply:
When we can avoid them (e.g., `StateUpdater`) I think this is better to document expanded types with primitive and object types. In this case, `Collection` is an important concept so I would recommend keeping it and adding a link.

The source is fine to start, but since it may move around, the link can break easily. An even better way would be to document custom types (maybe under a section **TypeScript**).

Reply:
Yes, that'll be nice!